internal/types/finality_providers.go

```diff
@@ -35,5 +42,20 @@ func NewFinalityProviders(filePath string) ([]FinalityProviderDetails, error) {
 		return nil, err
 	}
 
-	return finalityProviders.FinalityProviders, nil
+	// Convert FinalityProviderFromFile to FinalityProviderDetails
+	var finalityProviderDetails []FinalityProviderDetails
+	for _, fp := range finalityProviders.FinalityProviders {
+		btcPk := fp.BtcPk
+		if fp.EotsPk != "" {
```

Comment:
Let's note that `eots_pk` is the default value and if it is not set, then `btc_pk` is used.